cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewApplicationManager.java

```diff
@@ -0,0 +1,90 @@
+import co.cask.cdap.common.namespace.NamespacedLocationFactory;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.registry.UsageRegistry;
+import co.cask.cdap.data2.security.Impersonator;
+import co.cask.cdap.internal.app.deploy.pipeline.ApplicationRegistrationStage;
+import co.cask.cdap.internal.app.deploy.pipeline.ApplicationVerificationStage;
+import co.cask.cdap.internal.app.deploy.pipeline.CreateDatasetInstancesStage;
+import co.cask.cdap.internal.app.deploy.pipeline.DeployDatasetModulesStage;
+import co.cask.cdap.internal.app.deploy.pipeline.LocalArtifactLoaderStage;
+import co.cask.cdap.internal.app.deploy.pipeline.ProgramGenerationStage;
+import co.cask.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import co.cask.cdap.pipeline.Pipeline;
+import co.cask.cdap.pipeline.PipelineFactory;
+import co.cask.cdap.security.authorization.AuthorizerInstantiator;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+
+/**
+ * This class is concrete implementation of {@link Manager} that deploys an Preview Application.
+ *
+ * @param <I> Input type.
+ * @param <O> Output type.
+ */
+public class PreviewApplicationManager<I, O> implements Manager<I, O> {
+  private final PipelineFactory pipelineFactory;
+  private final NamespacedLocationFactory namespacedLocationFactory;
+  private final CConfiguration cConf;
+  private final Store store;
+  private final DatasetFramework datasetFramework;
+  private final DatasetFramework inMemoryDatasetFramework;
+  private final UsageRegistry usageRegistry;
+  private final ArtifactRepository artifactRepository;
+  private final AuthorizerInstantiator authorizerInstantiator;
+  private final Impersonator impersonator;
+
+  @Inject
+  PreviewApplicationManager(CConfiguration configuration, PipelineFactory pipelineFactory,
+                            NamespacedLocationFactory namespacedLocationFactory,
+                            Store store, DatasetFramework datasetFramework,
+                            @Named("datasetMDS") DatasetFramework inMemoryDatasetFramework,
+                            UsageRegistry usageRegistry, ArtifactRepository artifactRepository,
+                            AuthorizerInstantiator authorizerInstantiator, Impersonator impersonator) {
+    this.cConf = configuration;
+    this.namespacedLocationFactory = namespacedLocationFactory;
+    this.pipelineFactory = pipelineFactory;
+    this.store = store;
+    this.datasetFramework = datasetFramework;
+    this.inMemoryDatasetFramework = inMemoryDatasetFramework;
+    this.usageRegistry = usageRegistry;
+    this.artifactRepository = artifactRepository;
+    this.authorizerInstantiator = authorizerInstantiator;
+    this.impersonator = impersonator;
+  }
+
+  @Override
+  public ListenableFuture<O> deploy(I input) throws Exception {
+    Pipeline<O> pipeline = pipelineFactory.getPipeline();
```

Comment:
Please refer to the latest `LocalApplicationManager`. You'll at least need the `pipeline.setFinally(new DeploymentCleanupStage());`